plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/FileTree.tsx

```diff
@@ -23,13 +24,49 @@ const FileTree: React.FC<Props> = ({
   onDatasetAdd,
   onOpenDetails,
 }) => {
-  const [selectedID, select] = useState<string>();
+  const [selectedItem, selectItem] = useState<DataCatalogItem>();
   const [expandedFolders, setExpandedFolders] = useState<{ id?: string; name?: string }[]>([]);
 
-  const handleSelect = useCallback((dataID?: string) => {
-    select(dataID);
+  const handleSelect = useCallback((item?: DataCatalogItem) => {
+    selectItem(item);
   }, []);
 
+  useEffect(() => {
+    postMsg({ action: "catalogModalOpen" }); // Needed to trigger sending client storage data from Sidebar
+  }, []);
+
```

Comment:
I don't think this is really good.
Already this filetree has become really heavy (we will talk about that later). But this catalogModalOpen is going to send the datacatalog the catalog on open twice. Not sure why it is needed here.

Reply:
I think you might want to either make a new action, or update the current catalogModalOpen action to be able to avoid sending the catalog each time.
Using this action for dataset and the folder keys separately is fine because they can be empty and/or small. But I feel like the catalog should be avoided except for when necessary

Reply:
You've already did something like this here:
https://github.com/eukarya-inc/reearth-plateauview/blob/a8c27dcbfaed3bbaf7d6213e154d714cb4ff5a9c/plugin/web/extensions/sidebar/modals/datacatalog/components/hooks.ts#L36
I feel like you are mixing between `initDataCatalog` and `catalogModalOpen`

Reply:
Nour that is exactly what I am tlaking about. Because we do that initDataCatalog, which receives catalog, among other things, I don't see why we need this catalogModalOpen in the Filetree.tsx file. Can you explain why this is desirable here?

Reply:
Because, otherwise the expandedFolders and dataset variables will not be received even if I get them from a new action in the sidebar.